src/openrct2/EditorObjectSelectionSession.cpp

```diff
@@ -611,13 +611,14 @@ bool window_editor_object_selection_select_object(uint8_t isMasterObject, int32_
 
 bool editor_check_object_group_at_least_one_selected(ObjectType checkObjectType)
 {
-    int32_t numObjects = static_cast<int32_t>(object_repository_get_items_count());
+    auto numObjects = object_repository_get_items_count();
```

Comment:
if you made this `std::min(object_repository_get_items_count(), _objectSelectionFlags.size())` then you don't need to add it to the other if. This feels like it will just cause the issue to crop up elsewhere though.

Reply:
That would work, but it would also make the code harder to understand IMO.

I'm not too fussed either way, but I would like to hear from another team member.

Reply:
Its a bit long but logically speaking that should be preferred over adding a new condition in a loop.